tensorflow-examples/src/main/java/org/tensorflow/model/examples/cnn/fastrcnn/FasterRcnnInception.java

```diff
@@ -287,35 +287,59 @@ public static void main(String[] params) {
+                            try (TFloat32 boxes = TFloat32.tensorOf(boxesShape)) {
+                                //batch size of 1
+                                boxes.setFloat(1, 0, 0, 0);
+                                for (FloatNdArray floatNdArray : boxArray) {
+                                    boxes.set(floatNdArray, 0, boxCount);
+                                    boxCount++;
                                 }
-                                //TODO tf.image.encodeJpeg
-                                ImageIO.write(bufferedImage, "jpg", new File(outputImagePath));
-                            } catch (IOException e) {
-                                System.err.println("Exception with writing image " + e.getMessage());
+                                //convert the 4D input image to normalised 0.0f - 1.0f
+                                //Draw bounding boxes using boxes tensor and list of colors
+                                //multiply by 255 then reshape and recast to TUint8 3D tensor
+                                //Create JPEG from the Tensor with quality of 100%
+                                EncodeJpeg.Options jpgOptions = EncodeJpeg.quality(100L);
+                                //output the JPEG to file
+                                WriteFile writeFile = tf.io.writeFile(tf.constant(outputImagePath),
+                                        tf.image.encodeJpeg(
+                                                tf.dtypes.cast(tf.reshape(
+                                                        tf.math.mul(
+                                                                tf.image.drawBoundingBoxes(tf.math.div(
+                                                                        tf.dtypes.cast(tf.constant(reshapeTensor),
+                                                                                TFloat32.class),
+                                                                        tf.constant(255.0f)
+                                                                        ),
+                                                                        tf.constant(boxes), colors),
+                                                                tf.constant(255.0f)
+                                                        ),
+                                                        tf.array(
+                                                                outputImage.shape().asArray()[0],
+                                                                outputImage.shape().asArray()[1],
+                                                                outputImage.shape().asArray()[2]
+                                                        )
+                                                ), TUint8.class),
+                                                jpgOptions));
+                                runner = s.runner();
+                                runner.addTarget(writeFile).run();
```

Comment:
This can be `s.run(writeFile)` if it's left this way, though if you do the placeholder conversion I suggested then you'll need to feed in the two tensors before adding the target.

Reply:
Used placeholders